File: 7kyu 1/evenLast.js
/*
|--------------------------------------------------------------------------
| Evens times last
|--------------------------------------------------------------------------
|
*/

function evenLast(numbers) {
   if (numbers.length < 1) {
      return 0;
   } else {
      let res = 0;
      for (let i in numbers) {
         if (i % 2 == 0) {
            res += numbers[i];
         }
      }
      return res * numbers.pop();
   }
}


const evenLast = n => n.reduce((p, c, i) => i % 2 ? p : p + c, 0) * n[n.length - 1] || 0


const evenLast = numbers => {
   const even = numbers.filter((a, i) => i % 2 == 0);
   return even.reduce((acc, val) => val * numbers[numbers.length - 1] + acc, 0);
}


console.log(evenLast([2, 3, 4, 5]), 30)
console.log(evenLast([]), 0)

File: Python/7py 1/even_last.py
"""
|--------------------------------------------------------------------------
| Evens times last
|--------------------------------------------------------------------------
|
"""

def even_last(numbers): 
   res = 0
   for i in range(len(numbers)):
      if i % 2 == 0:
         res += numbers[i]
   return res * numbers[-1] if numbers else 0


def even_last(n): 
   return 0 if not n else sum(n[::2])*n[-1]


def even_last(numbers): 
   return bool(numbers) and sum(numbers[::2]) * numbers[-1]


def even_last(num): 
   return num and sum(num[::2]) * num[-1] or 0
  

print(even_last([2, 3, 4, 5]), 30)
print(even_last([]), 0)